sentry-ruby/lib/sentry/transport.rb

```diff
@@ -82,6 +82,26 @@ def serialize_envelope(envelope)
           result = item.to_s
         end
 
+        if result.bytesize > Event::MAX_SERIALIZED_PAYLOAD_SIZE
+          if single_exceptions = item.payload.dig(:exception, :values)
+            single_exceptions.each do |single_exception|
+              traces = single_exception.dig(:stacktrace, :frames)
+              if traces && traces.size > 0
+                single_exception.delete(:stacktrace)
+              end
+            end
+          elsif single_exceptions = item.payload.dig("exception", "values")
+            single_exceptions.each do |single_exception|
+              traces = single_exception.dig("stacktrace", "frames")
+              if traces && traces.size > 0
+                single_exception.delete("stacktrace")
+              end
+            end
+          end
+
+          result = item.to_s
+        end
+
```

Comment:
I can't comment lower so I just leave it here: we also need to update the message form:

```
is still oversized without breadcrumbs
```

to

```
is still oversized without breadcrumbs and stracktrace
```